packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -71,12 +106,266 @@ export class IamPolicyChecksWebview extends VueWebview {
         })
     }
 
-    // TODO: Add functionality for the three checks
-    public async validatePolicy(documentType: string, policyType: string) {
-        this.client /* Do something */
+    /* Uses A2 SDK and Python CLI tools to call ValidatePolicy API
+     * Responses are exposed in the Webview, including errors
+     * Diagnostic objects are created to expose findings, which appear in the problems panel
+     */
+    public async validatePolicy(documentType: string, policyType: string, cfnParameterPath?: string) {
+        const document = IamPolicyChecksWebview.editedDocumentFileName
+        validatePolicyDiagnosticCollection.clear()
+        const diagnostics: vscode.Diagnostic[] = []
+        switch (documentType) {
+            case PolicyChecksDocumentType.JSON_POLICY_LANGUAGE: {
+                this.client.validatePolicy(
```

Comment:
Q: do we do a basic JSON linting before sending the request?

Reply:
VS Code provides it out of the box, but ValidatePolicy does very basic linting as well.